One Integration test broken locally `TestSnapshotHardDelete.test__postgres__snapshot_hard_delete`

### Describe the bug
Hello :wave:,

I followed the [CONTRIBUTING.md](https://github.com/dbt-labs/dbt/blob/develop/CONTRIBUTING.md) guide to run `dbt` in development, here the steps I did:

- use `pyenv` and `virtualenv` to manage my python env.
- Instal all python dependencies (`pip install -r dev-requirements.txt -r editable-requirements.txt
`)
- init the database (`make setup-db`)
- launch the unit-tests :heavy_check_mark: 
- launch the integration tests :red_circle: 

`004_simple_snapshot_test` integration test failed:

```
_______________________________________________________________________ TestSnapshotHardDelete.test__postgres__snapshot_hard_delete _______________________________________________________________________

self = <test_simple_snapshot.TestSnapshotHardDelete testMethod=test__postgres__snapshot_hard_delete>

    @use_profile('postgres')
    def test__postgres__snapshot_hard_delete(self):
        self.run_sql_file('seed_pg.sql')
>       self._test_snapshot_hard_delete()

test/integration/004_simple_snapshot_test/test_simple_snapshot.py:838: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test/integration/004_simple_snapshot_test/test_simple_snapshot.py:867: in _test_snapshot_hard_delete
    self._snapshot_and_assert_invalidated()
test/integration/004_simple_snapshot_test/test_simple_snapshot.py:909: in _snapshot_and_assert_invalidated
    self.assertGreaterEqual(result[-1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime)
E   AssertionError: datetime.datetime(2021, 9, 26, 6, 8, 10, 84187, tzinfo=<UTC>) not greater than or equal to datetime.datetime(2021, 9, 26, 8, 8, 9, 594456, tzinfo=<UTC>)
``` 

FYI, I launched the test at 9:26 UTC.
Seems like `result[-1].astimezone(pytz.UTC)` doesn't return the right UTC time.

----------------
I did few tests:
`result[-1].replace(tzinfo=pytz.UTC)` fix the issue home.

If we took a look to [astimezone doc](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone)
```
Changed in version 3.6: The astimezone() method can now be called on naive instances that are presumed to represent system local time.
```

### Steps To Reproduce
Not sure to know how to reproduce the BUG. Root cause if probably linked to my local environment :confused: .
Please note my environment variable `TZ` is equal to `Europe/Paris` .
Same for PG Timezone:
```
SHOW TIMEZONE;
Europe/Paris
```

- Install DBT in dev following the [CONTRIBUTING.md](https://github.com/dbt-labs/dbt/blob/develop/CONTRIBUTING.md) guide
- To launch the test:
```
 python -m pytest -m profile_postgres test/integration/004_simple_snapshot_test/test_simple_snapshot.py::TestSnapshotHardDelete::test__postgres__snapshot_hard_delete
```


### Expected behavior
Integration tests to pass.

### System information
**Which database are you using dbt with?**
- [X] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)


**The output of `dbt --version`:**
```
installed version: 0.21.0-rc1
   latest version: 0.20.2

Your version of dbt is ahead of the latest release!

Plugins:
  - postgres: 0.21.0rc1
  - snowflake: 0.21.0rc1
  - redshift: 0.21.0rc1
  - bigquery: 0.21.0rc1
```

**The operating system you're using:**
Ubuntu 20.04.2 LTS

**The output of `python --version`:**
Python 3.8.11

### Additional context
I can contribute to fix this bug if needed :slightly_smiling_face: .

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,7 +14,6 @@ Contributors:
 
 - [@dave-connors-3](https://github.com/dave-connors-3) ([#3920](https://github.com/dbt-labs/dbt/issues/3920))
 
-
 ## dbt 0.21.0 (Release TBD)
 
 ### Fixes
@@ -22,6 +21,10 @@ Contributors:
 
 ### Under the hood
 - Bump artifact schema versions for 0.21.0 ([#3945](https://github.com/dbt-labs/dbt/pull/3945))
+
+Contributors:
+
+- [@kadero](https://github.com/kadero) ([#3952](https://github.com/dbt-labs/dbt/issues/3952))
 
 ## dbt 0.21.0rc1 (September 20, 2021)
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ Contributors:
 
 - [@dave-connors-3](https://github.com/dave-connors-3) ([#3920](https://github.com/dbt-labs/dbt/issues/3920))
 
+
 ## dbt 0.21.0 (Release TBD)
 
 ### Fixes

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,7 @@
 Contributors:
 
 - [@dave-connors-3](https://github.com/dave-connors-3) ([#3920](https://github.com/dbt-labs/dbt/issues/3920))
-
+- [@kadero](https://github.com/kadero) ([#3952](https://github.com/dbt-labs/dbt/issues/3952))
 
 ## dbt 0.21.0 (Release TBD)
 
@@ -22,10 +22,6 @@ Contributors:
 
 ### Under the hood
 - Bump artifact schema versions for 0.21.0 ([#3945](https://github.com/dbt-labs/dbt/pull/3945))
-
-Contributors:
-
-- [@kadero](https://github.com/kadero) ([#3952](https://github.com/dbt-labs/dbt/issues/3952))
 
 ## dbt 0.21.0rc1 (September 20, 2021)
 

--- a/test/integration/004_simple_snapshot_test/test_simple_snapshot.py
+++ b/test/integration/004_simple_snapshot_test/test_simple_snapshot.py
@@ -906,7 +906,7 @@ class TestSnapshotHardDelete(DBTIntegrationTest):
         for result in snapshotted[10:]:
             # result is a tuple, the dbt_valid_to column is the latest
             self.assertIsInstance(result[-1], datetime)
-            self.assertGreaterEqual(result[-1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime)
+            self.assertGreaterEqual(result[-1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime)
 
     def _revive_records(self):
         database = self.default_database
@@ -946,7 +946,7 @@ class TestSnapshotHardDelete(DBTIntegrationTest):
         for result in invalidated_records:
             # result is a tuple, the dbt_valid_to column is the latest
             self.assertIsInstance(result[1], datetime)
-            self.assertGreaterEqual(result[1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime)
+            self.assertGreaterEqual(result[1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime)
 
         # records which weren't revived (id != 10, 11)
         revived_records = self.run_sql(
@@ -968,5 +968,5 @@ class TestSnapshotHardDelete(DBTIntegrationTest):
             self.assertIsInstance(result[1], datetime)
             # there are milliseconds (part of microseconds in datetime objects) in the
             # invalidated_snapshot_datetime and not in result datetime so set the microseconds to 0
-            self.assertGreaterEqual(result[1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime.replace(microsecond=0))
+            self.assertGreaterEqual(result[1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime.replace(microsecond=0))
             self.assertIsNone(result[2])

